### PR TITLE
Custom domains

### PR DESCRIPTION
**Pull request by jorgeblacio**

### CODE REVIEW

File: src/main/kotlin/com/criptext/mail/scenes/settings/workers/CheckCustomDomainWorker.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    : BackgroundWorker<SettingsResult.CheckCustomDomain> {
+
+    override val canBeParallelized = true
+
+    override fun catchException(ex: Exception): SettingsResult.CheckCustomDomain {
+        return SettingsResult.CheckCustomDomain.Failure(createErrorMessage(ex))
+    }
+
+    override fun work(reporter: ProgressReporter<SettingsResult.CheckCustomDomain>): SettingsResult.CheckCustomDomain? {
+        val customDomain = customDomainDao.getAll()
```
Comment (danieltigse) [inline src/main/kotlin/com/criptext/mail/scenes/settings/workers/CheckCustomDomainWorker.kt]: You can add this query into the existing worker in settings

File: src/main/res/layout/custom_domain_item.xml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                android:textSize="14sp"
+                android:layout_weight="0.8"/>
+
+            <ImageView
+                android:id="@+id/imageViewTrashDomain"
+                android:layout_width="40dp"
+                android:layout_height="40dp"
+                android:layout_marginEnd="7dp"
+                android:src="@drawable/trash"
+                android:tint="#BBBBBB" />
```
Comment (danieltigse) [inline src/main/res/layout/custom_domain_item.xml]: Create a resource for it. Check other places where you use it.

File: src/main/res/values-es/strings.xml [hunk 3]
```diff
@@ -216,6 +217,7 @@
   <string name="username_invalid_error">Nombre de Usuario inválido.</string>
   <string name="username_length_error">Al menos 3 caracteres. Debe comenzar con letras A-Z.</string>
   <string name="email_invalid_error">Dirección de correo inválida.</string>
+  <string name="domain_invalid_error">Invalid domain.</string>
```
Comment (danieltigse) [inline src/main/res/values-es/strings.xml]: Add Spanish translations

File: src/main/kotlin/com/criptext/mail/scenes/settings/workers/CheckCustomDomainWorker.kt (hunk 1 above)
Comment (jorgeblacio) [reply src/main/kotlin/com/criptext/mail/scenes/settings/workers/CheckCustomDomainWorker.kt]: This is fixed on a future PR.

File: src/main/kotlin/com/criptext/mail/scenes/settings/custom_domain/workers/LoadDomainWorker.kt [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    private val apiClient = CustomDomainAPIClient(httpClient, activeAccount.jwt)
+
+    override fun catchException(ex: Exception): CustomDomainResult.LoadDomain {
+        return CustomDomainResult.LoadDomain.Failure(UIMessage(R.string.unknown_error, arrayOf(ex.toString())))
+    }
+
+    override fun work(reporter: ProgressReporter<CustomDomainResult.LoadDomain>): CustomDomainResult.LoadDomain? {
+        val operation = workOperation()
+
+        val sessionExpired = HttpErrorHandlingHelper.didFailBecauseInvalidSession(operation)
```
[the remark this answers is not in the document]
Comment (jorgeblacio) [reply src/main/kotlin/com/criptext/mail/scenes/settings/custom_domain/workers/LoadDomainWorker.kt]: This is fixed in future pr

[the remark this answers is not in the document]
Comment (jorgeblacio) [reply src/main/kotlin/com/criptext/mail/scenes/settings/custom_domain/workers/LoadDomainWorker.kt]: This is fixed in next PR

**Pull request merged**